\doxysection{File List}
Here is a list of all files with brief descriptions\+:\begin{DoxyCompactList}
\item\contentsline{section}{script/\mbox{\hyperlink{create__geno__haplo__data_8cpp}{create\+\_\+geno\+\_\+haplo\+\_\+data.\+cpp}} \\*This program generates genotypic and haplotypic data based on the parameters specified in an input file {\ttfamily parameters.\+txt}. The results are saved in two CSV files\+: {\ttfamily genotypes.\+csv} and {\ttfamily haplotypes.\+csv} }{\pageref{create__geno__haplo__data_8cpp}}{}
\item\contentsline{section}{script/\mbox{\hyperlink{generate__genotypes_8h}{generate\+\_\+genotypes.\+h}} \\*Defines the function {\ttfamily generate\+\_\+genotypes} for generating genotypes from haplotypes }{\pageref{generate__genotypes_8h}}{}
\item\contentsline{section}{script/\mbox{\hyperlink{script_2generate__haplotypes_8h}{generate\+\_\+haplotypes.\+h}} \\*This header file defines the function {\ttfamily generate\+\_\+haplotypes}, which generates random haplotypes for a given population of individuals }{\pageref{script_2generate__haplotypes_8h}}{}
\item\contentsline{section}{script/\mbox{\hyperlink{script_2read__parameters_8h}{read\+\_\+parameters.\+h}} \\*Defines the function {\ttfamily read\+\_\+parameters} to read configuration parameters from a specified text file }{\pageref{script_2read__parameters_8h}}{}
\item\contentsline{section}{script/\mbox{\hyperlink{save__csv_8h}{save\+\_\+csv.\+h}} \\*Defines the function {\ttfamily save\+\_\+csv} for saving a 2D vector of integers into a CSV file }{\pageref{save__csv_8h}}{}
\item\contentsline{section}{src/alter\+\_\+data/\mbox{\hyperlink{alter__data_8cpp}{alter\+\_\+data.\+cpp}} \\*This program modifies genotype data by introducing missing values based on parameters specified in an input file {\ttfamily parameters.\+txt} }{\pageref{alter__data_8cpp}}{}
\item\contentsline{section}{src/alter\+\_\+data/\mbox{\hyperlink{introduce__missing__data_8h}{introduce\+\_\+missing\+\_\+data.\+h}} \\*Defines a function to introduce missing data into a genotype matrix }{\pageref{introduce__missing__data_8h}}{}
\item\contentsline{section}{src/alter\+\_\+data/\mbox{\hyperlink{read__genotypes_8h}{read\+\_\+genotypes.\+h}} \\*This header file defines a function to read genotype data from a text file. The genotype data is stored in a 2D vector of integers. Each row in the file corresponds to a row in the matrix, and non-\/numeric characters are ignored }{\pageref{read__genotypes_8h}}{}
\item\contentsline{section}{src/alter\+\_\+data/\mbox{\hyperlink{src_2alter__data_2read__parameters_8h}{read\+\_\+parameters.\+h}} \\*Defines a function to read simulation parameters from a text file }{\pageref{src_2alter__data_2read__parameters_8h}}{}
\item\contentsline{section}{src/alter\+\_\+data/\mbox{\hyperlink{write__genotypes_8h}{write\+\_\+genotypes.\+h}} \\*Defines a function to write genotype data to a specified output file }{\pageref{write__genotypes_8h}}{}
\item\contentsline{section}{src/infer\+\_\+haplo/\mbox{\hyperlink{assign__haplotypes_8h}{assign\+\_\+haplotypes.\+h}} }{\pageref{assign__haplotypes_8h}}{}
\item\contentsline{section}{src/infer\+\_\+haplo/\mbox{\hyperlink{calcul__proba__genotypes_8h}{calcul\+\_\+proba\+\_\+genotypes.\+h}} \\*Calculates the probabilities of genotypes based on haplotype frequencies }{\pageref{calcul__proba__genotypes_8h}}{}
\item\contentsline{section}{src/infer\+\_\+haplo/\mbox{\hyperlink{calcul_average_hamming_distance_8h}{calcul\+Average\+Hamming\+Distance.\+h}} \\*Computes the normalized average Hamming distance between two sets of haplotypes }{\pageref{calcul_average_hamming_distance_8h}}{}
\item\contentsline{section}{src/infer\+\_\+haplo/\mbox{\hyperlink{calcul_hamming_distance_8h}{calcul\+Hamming\+Distance.\+h}} \\*Computes the Hamming distance between two haplotypes }{\pageref{calcul_hamming_distance_8h}}{}
\item\contentsline{section}{src/infer\+\_\+haplo/\mbox{\hyperlink{distinct__haplotypes_8h}{distinct\+\_\+haplotypes.\+h}} }{\pageref{distinct__haplotypes_8h}}{}
\item\contentsline{section}{src/infer\+\_\+haplo/\mbox{\hyperlink{estimation__esperance_8h}{estimation\+\_\+esperance.\+h}} \\*Computes the log-\/likelihood from genotype probabilities }{\pageref{estimation__esperance_8h}}{}
\item\contentsline{section}{src/infer\+\_\+haplo/\mbox{\hyperlink{explain__genotype_8h}{explain\+\_\+genotype.\+h}} \\*Checks whether a pair of haplotypes can explain a given genotype }{\pageref{explain__genotype_8h}}{}
\item\contentsline{section}{src/infer\+\_\+haplo/\mbox{\hyperlink{src_2infer__haplo_2generate__haplotypes_8h}{generate\+\_\+haplotypes.\+h}} \\*Generates possible haplotype pairs for a given genotype and filters unique haplotypes }{\pageref{src_2infer__haplo_2generate__haplotypes_8h}}{}
\item\contentsline{section}{src/infer\+\_\+haplo/\mbox{\hyperlink{genotype__reader_8h}{genotype\+\_\+reader.\+h}} \\*Provides functionality for reading genotypes from a CSV file }{\pageref{genotype__reader_8h}}{}
\item\contentsline{section}{src/infer\+\_\+haplo/\mbox{\hyperlink{haplotype__reader_8h}{haplotype\+\_\+reader.\+h}} }{\pageref{haplotype__reader_8h}}{}
\item\contentsline{section}{src/infer\+\_\+haplo/\mbox{\hyperlink{haplotype__reordering_8h}{haplotype\+\_\+reordering.\+h}} }{\pageref{haplotype__reordering_8h}}{}
\item\contentsline{section}{src/infer\+\_\+haplo/\mbox{\hyperlink{infer__haplo_8cpp}{infer\+\_\+haplo.\+cpp}} }{\pageref{infer__haplo_8cpp}}{}
\item\contentsline{section}{src/infer\+\_\+haplo/\mbox{\hyperlink{initialisation__frequences_8h}{initialisation\+\_\+frequences.\+h}} \\*Initializes the frequencies of distinct haplotypes }{\pageref{initialisation__frequences_8h}}{}
\item\contentsline{section}{src/infer\+\_\+haplo/\mbox{\hyperlink{log__writer_8h}{log\+\_\+writer.\+h}} \\*Writes log information to a specified log file }{\pageref{log__writer_8h}}{}
\item\contentsline{section}{src/infer\+\_\+haplo/\mbox{\hyperlink{maximisation_8h}{maximisation.\+h}} \\*Updates haplotype frequencies in the maximization step of the Expectation-\/\+Maximization (EM) algorithm }{\pageref{maximisation_8h}}{}
\item\contentsline{section}{src/infer\+\_\+haplo/\mbox{\hyperlink{sort__haplotypes_8h}{sort\+\_\+haplotypes.\+h}} \\*Sorts haplotypes by their frequency in descending order }{\pageref{sort__haplotypes_8h}}{}
\item\contentsline{section}{src/infer\+\_\+haplo/\mbox{\hyperlink{two_numbers_in_genotypes_8h}{two\+Numbers\+In\+Genotypes.\+h}} \\*Counts the total occurrences of the value {\ttfamily 2} in a genotypes CSV file }{\pageref{two_numbers_in_genotypes_8h}}{}
\item\contentsline{section}{src/infer\+\_\+haplo/\mbox{\hyperlink{vector__hash_8h}{vector\+\_\+hash.\+h}} \\*Custom hash function for {\ttfamily std\+::vector\texorpdfstring{$<$}{<}int\texorpdfstring{$>$}{>}} to be used in hash-\/based containers }{\pageref{vector__hash_8h}}{}
\item\contentsline{section}{src/infer\+\_\+missing\+\_\+data/\mbox{\hyperlink{hamming__distance_8h}{hamming\+\_\+distance.\+h}} }{\pageref{hamming__distance_8h}}{}
\item\contentsline{section}{src/infer\+\_\+missing\+\_\+data/\mbox{\hyperlink{impute__missing__data_8h}{impute\+\_\+missing\+\_\+data.\+h}} }{\pageref{impute__missing__data_8h}}{}
\item\contentsline{section}{src/infer\+\_\+missing\+\_\+data/\mbox{\hyperlink{infer__missing__data_8cpp}{infer\+\_\+missing\+\_\+data.\+cpp}} \\*Implements the main function for genotype imputation using the k-\/nearest neighbors (KNN) approach }{\pageref{infer__missing__data_8cpp}}{}
\item\contentsline{section}{src/infer\+\_\+missing\+\_\+data/\mbox{\hyperlink{read__genotypes__knn_8h}{read\+\_\+genotypes\+\_\+knn.\+h}} \\*Reads genotype data from a CSV file, handling missing values }{\pageref{read__genotypes__knn_8h}}{}
\item\contentsline{section}{src/infer\+\_\+missing\+\_\+data/\mbox{\hyperlink{read__parameters__knn_8h}{read\+\_\+parameters\+\_\+knn.\+h}} \\*Reads the parameter file to extract the value of {\ttfamily k} for KNN imputation }{\pageref{read__parameters__knn_8h}}{}
\item\contentsline{section}{src/infer\+\_\+missing\+\_\+data/\mbox{\hyperlink{write__genotypes__knn_8h}{write\+\_\+genotypes\+\_\+knn.\+h}} \\*Writes the imputed genotype data to a CSV file, handling missing values }{\pageref{write__genotypes__knn_8h}}{}
\item\contentsline{section}{toy\+\_\+example\+\_\+alter\+\_\+data/\mbox{\hyperlink{toy__example__alter__data_2launch__toy__example_8py}{launch\+\_\+toy\+\_\+example.\+py}} }{\pageref{toy__example__alter__data_2launch__toy__example_8py}}{}
\item\contentsline{section}{toy\+\_\+example\+\_\+create\+\_\+geno\+\_\+haplo\+\_\+data/\mbox{\hyperlink{toy__example__create__geno__haplo__data_2launch__toy__example_8py}{launch\+\_\+toy\+\_\+example.\+py}} }{\pageref{toy__example__create__geno__haplo__data_2launch__toy__example_8py}}{}
\item\contentsline{section}{toy\+\_\+example\+\_\+infer\+\_\+haplo/\mbox{\hyperlink{toy__example__infer__haplo_2launch__toy__example_8py}{launch\+\_\+toy\+\_\+example.\+py}} }{\pageref{toy__example__infer__haplo_2launch__toy__example_8py}}{}
\item\contentsline{section}{toy\+\_\+example\+\_\+infer\+\_\+missing\+\_\+data/\mbox{\hyperlink{toy__example__infer__missing__data_2launch__toy__example_8py}{launch\+\_\+toy\+\_\+example.\+py}} }{\pageref{toy__example__infer__missing__data_2launch__toy__example_8py}}{}
\end{DoxyCompactList}
